services/director-v2/tests/unit/with_dbs/test_modules_redis.py

```diff
@@ -0,0 +1,279 @@
+from fastapi import FastAPI
+from redis.asyncio import Redis
+from redis.asyncio.lock import Lock
+from redis.exceptions import LockError
+from settings_library.redis import RedisSettings
+from simcore_service_director_v2.modules import redis
+from simcore_service_director_v2.modules.redis import (
+    EXTEND_TASK_ATTR_NAME,
+    DockerNodeId,
+    RedisLockManager,
+    auto_release,
+)
+
+pytest_simcore_core_services_selection = [
+    "redis",
+]
+
+# UTILS
+
+
+class MockLocksPerNodeProvider:
+    def __init__(self, lock_per_node: int) -> None:
+        self.lock_per_node = lock_per_node
+
+    async def get(self, *args, **kwargs):
+        return self.lock_per_node
+
+
+# FIXTURES
+
+
+@pytest.fixture
+async def minimal_app(redis_settings: RedisSettings) -> AsyncIterable[FastAPI]:
+    app = FastAPI()
+
+    # add expected redis_settings
+    app.state.settings = Namespace()
+    app.state.settings.REDIS = redis_settings
+
+    # setup redis module
+    redis.setup(app)
+
+    async with LifespanManager(app):
+        yield app
+
+
+@pytest.fixture
+async def redis_lock_manager(minimal_app: FastAPI) -> AsyncIterable[RedisLockManager]:
+    redis_lock_manger = RedisLockManager.instance(minimal_app)
+    await redis_lock_manger.redis.flushall()
+    yield redis_lock_manger
+    await redis_lock_manger.redis.flushall()
+
+
+@pytest.fixture
+def docker_node_id(faker: Faker) -> str:
+    return faker.uuid4()
+
+
+# TESTS
+
+
+async def test_lock_working_as_expected(
+    redis_lock_manager: RedisLockManager, docker_node_id
+) -> None:
+    lock = redis_lock_manager.redis.lock(docker_node_id)
+
+    lock_acquired = await lock.acquire(blocking=False)
+    assert lock_acquired
+    assert await lock.locked() is True
+
+    await lock.release()
+    assert await lock.locked() is False
+
+    with pytest.raises(LockError):
+        await lock.release()
+
+
+async def test_two_lock_instances(
+    redis_lock_manager: RedisLockManager, docker_node_id
+) -> None:
+    # NOTE: this test show cases how the locks work
+    # you have to acquire the lock from the same istance
+    # in order to avoid tricky situations
+
+    lock = redis_lock_manager.redis.lock(docker_node_id)
+
+    lock_acquired = await lock.acquire(blocking=False)
+    assert lock_acquired
+    assert await lock.locked() is True
+
+    # we get a different instance
+    second_lock = redis_lock_manager.redis.lock(docker_node_id)
+    assert await second_lock.locked() is True
+
+    # cannot release lock form different instance!
+    with pytest.raises(LockError):
+        await second_lock.release()
+
+    assert await lock.locked() is True
+    # NOTE: this is confusing! One woudl expect the second lock to be unlocked
+    # but it actually is True
+    assert await second_lock.locked() is True
+
+    await lock.release()
+    assert await lock.locked() is False
+    # NOTE: apparently it mirrors the first lock instance!
+    assert await second_lock.locked() is False
```

Comment:
These notes suggest that you are getting results you did not expect. Please check with @sanderegg how this library works. He has some experience with it.